### ISSUE TITLE
root: true not behaving correctly for config

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#3109]
Version: `master`
Parser: `Default`

Command to run: `npm test` on master

Scenario: I have `eslint` folder and also have an `.eslintrc` file alongside it. when I run the `npm test` command on `eslint` folder it also grabs the `.eslintrc` file from outside eslint folder. Important to note that our `.eslintrc` file has `root:true` so that means it should not grab the file from outside.

Output:

```sh
$ DEBUG=eslint:* npm test

> eslint@1.0.0-rc-1 test c:\Users\gs025879\Documents\webstrom\eslint
> node Makefile.js test

Validating Makefile.js
Tue, 21 Jul 2015 18:01:15 GMT eslint:cli Running on files
Tue, 21 Jul 2015 18:01:15 GMT eslint:cli-engine Processing ./Makefile.js
Tue, 21 Jul 2015 18:01:15 GMT eslint:cli-engine Linting ./Makefile.js
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Constructing config for c:\Users\gs025879\Documents\webstrom\eslint\Makefile.js
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Using .eslintrc and package.json files
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Loading c:\Users\gs025879\Documents\webstrom\eslint\.eslintrc
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Using c:\Users\gs025879\Documents\webstrom\eslint\.eslintrc
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Loading c:\Users\gs025879\Documents\webstrom\eslint\package.json
Tue, 21 Jul 2015 18:01:15 GMT eslint:config Loading c:\Users\gs025879\Documents\webstrom\.eslintrc
c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:134
                    throw e;
                          ^
Error: Cannot find module 'eslint-config-ideal'
Referenced from: c:\Users\gs025879\Documents\webstrom\.eslintrc
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at loadConfig (c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:97:48)
    at c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:128:46
    at Array.reduceRight (native)
    at loadConfig (c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:112:36)
    at getLocalConfig (c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:231:23)
    at Config.getConfig (c:\Users\gs025879\Documents\webstrom\eslint\lib\config.js:364:22)
Validating JSON Files
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by nzakas** [comment]
Looks like a bug, we should add some debug logging so it's easier to tell when `root: true` was found and when it was not.

**Comment by gyandeeps** [comment]
I think why its happening is before we can say that we are going to ignore a particular config file, we have to load the config. In my case during the load of that it blows up because I dont have `eslint-config-ideal` installed.
I think the way to fix this is as soon as you find a config file with `root:true`, then we should exit the loop. As of now we keep reading and loading configs.

Would this change makes sense @nzakas and @IanVS . If yes, then I will implement this change.

**Comment by IanVS** [comment]
The problem with exiting immediately is that there may be eslint configuration in a package.json and we'll as an .eslintrc in the same folder.  If you exit immediately, you could miss one of them.

**Comment by IanVS** [comment]
I think the best thing to do is move the check (https://github.com/eslint/eslint/blob/master/lib/config.js#L244) to before the config is loaded.

**Comment by gyandeeps** [comment]
That would work too. Sounds good to me. Will have thew PR ready :+1:
